src/views/BlogDetails.tsx

```diff
@@ -1,7 +1,40 @@
+/** @jsxImportSource @emotion/react */
+import { blogPosts } from 'data';
+import { useNavigate, useParams } from 'react-router-dom';
+import { NoMatch } from 'shared';
+import { cardContainer } from './BlogDetails.styles';
+
+interface Params {
+  id: string;
+}
+
 export const BlogDetails: React.FC = () => {
+  const navigate = useNavigate();
+  const { id } = useParams<keyof Params>() as Params;
+
+  const result = blogPosts.filter((blogPost) => blogPost.id === parseInt(id));
```

Comment:
Umjesto filtera mogao si koristiti find, pa onda na osnovu toga si mogao odraditi conditional rendering: npr.
`const result = blogPosts.find((blogPost) => blogPost.id === parseInt(id));

  return result ? (
    <div css={cardContainer}>
      <div>
        <h1>{result.title}</h1>
        <h2>{result.description}</h2>
        <p>{result.text}</p>
        <button onClick={() => navigate('/blog')}>Back</button>
      </div>
    </div>
  ) : (
    <NoMatch
      description={`Oops, there is no blog with the entered ID (${id})`}
      path="/blog"
      pathName="Go to Blog page"
    />
  );`

Reply:
Object.keys(result).length === 0 također ovo nema smisla, mogao si samo na array objektu u tvom slucaju results traziti length:
```
if (result.length === 0) {
    return (
      <NoMatch
        description={`Oops, there is no blog with the entered ID (${id})`}
        path="/blog"
        pathName="Go to Blog page"
      />
    );
  }
```